checkout-core/src/main/java/com/adyen/checkout/core/internal/data/model/JsonUtils.kt

```diff
@@ -95,4 +132,41 @@ object JsonUtils {
             }
         }
     }
+
+    /**
+     * Parses a [JSONArray] to a list of integers.
+     *
+     * @param jsonArray The JSONArray to be read.
+     * @return A [List] of integers, or null if the jsonArray was null.
+     */
+    @JvmStatic
+    fun parseOptIntegerList(jsonArray: JSONArray?): List<Int>? {
+        if (jsonArray == null) {
+            return null
+        }
+        val list: MutableList<Int> = ArrayList()
+        for (i in 0 until jsonArray.length()) {
+            val item = jsonArray.optInt(i, -1)
+            list.add(item)
+        }
+        return Collections.unmodifiableList(list)
+    }
+
+    /**
+     * Serializes a List of integers to a [JSONArray].
+     *
+     * @param intList The [List] of integers to be serialized.
+     * @return The populated [JSONArray]. Could be null.
+     */
+    @JvmStatic
+    fun serializeOptIntegerList(intList: List<Int?>?): JSONArray? {
+        if (intList == null) {
+            return null
+        }
+        return JSONArray().apply {
+            intList.filter { it != -1 }.forEach {
```

Comment:
Here if a value is -1 we skip it and if it's null we add it as null? What if we actually need to serialize -1 on purpose?